generators/heroku/index.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable consistent-return */
+const chalk = require('chalk');
+const fs = require('fs');
+const ChildProcess = require('child_process');
+const util = require('util');
+const os = require('os');
+const HerokuGenerator = require('generator-jhipster/generators/heroku');
+const constants = require('generator-jhipster/generators/generator-constants');
+const { getBase64Secret } = require('generator-jhipster/generators/utils');
+
+const execCmd = util.promisify(ChildProcess.exec);
+
+module.exports = class extends HerokuGenerator {
+    constructor(args, opts) {
+        super(args, { fromBlueprint: true, ...opts }); // fromBlueprint variable is important
+
+        const jhContext = (this.jhipsterContext = this.options.jhipsterContext);
+
+        if (!jhContext) {
+            this.error(`This is a JHipster blueprint and should be used only like ${chalk.yellow('jhipster --blueprint micronaut')}`);
+        }
+
+        this.configOptions = jhContext.configOptions || {};
+    }
+
+    /**
+     * build a generated application.
+     *
+     * @param {String} buildTool - maven | gradle
+     * @param {String} profile - dev | prod
+     * @param {Boolean} buildWar - build a war instead of a jar
+     * @param {Function} cb - callback when build is complete
+     * @returns {object} the command line and its result
+     */
+    buildApplication(buildTool, profile, buildWar, cb) {
```

Comment:
This function will be queued as a task at default priority.
So it will be called at least one time using arguments passed to the generator.

Reply:
You mean it will be executed nevertheless it is called?

Reply:
Yep. Pretty sure. Can you confirm it?

Reply:
Will check, but pretty sure the build is not executed when e.g. using git deploy (sure I would have noticed that). But will check.

Reply:
Reproduced by:
```
npm install atomfrede/generator-jhipster-micronaut#82-support-heroku-deployment
DEBUG=yeoman:generator jhipster heroku --blueprints micronaut
```

Output:
```
yeoman:generator Queueing jhipster-micronaut:heroku#buildApplication with options { cancellable: true, run: false, generatorReject: [Function], taskName: 'buildApplication' } +0ms
```

The fix is to override buildApplication in a super class:
```
const HerokuGeneratorOverride = class extends HerokuGenerator {
   buildApplication(){...}
}
module.exports = class extends HerokuGeneratorOverride {
}
```

Reply:
Interesting, thanks @mshima for that.